### PR TITLE
[asr->fortran] implement visit_StringConcat

### PR DESCRIPTION
**Pull request by khushi-411**

### CODE REVIEW

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -928,7 +928,7 @@ RUN(NAME assign_to3 LABELS gfortran llvm fortran)
 
 RUN(NAME conv_complex2real LABELS gfortran llvm wasm)
 
-RUN(NAME template_01 LABELS llvm wasm)
+RUN(NAME template_01 LABELS llvm wasm fortran)
```
Comment (Shaikh-Ubaid) [inline integration_tests/CMakeLists.txt:931]: As discussed here https://github.com/lfortran/lfortran/pull/2566#issuecomment-1751707211, could you only enable tests that are related to your PR changes. If this test is related to the PR changes, please describe how it is related.

**Review by Shaikh-Ubaid — COMMENTED** [review]
Thanks for this. I shared a comment below.

File: integration_tests/CMakeLists.txt (hunk 1 above)
Comment (khushi-411) [reply integration_tests/CMakeLists.txt]: Added new tests and removed this; thanks for pointing it out, @Shaikh-Ubaid!

PS: I also checked the test code, and the fortran backend-generated code are same.

File: integration_tests/string_26.f90 [hunk 2]
```diff
@@ -0,0 +1,8 @@
+program string26
+character(len=3) :: s1, s2
+character(len=6) :: s
+s1 = "abc"
+s2 = "def"
+s = s1//s2
+print *, s
```
Comment (Shaikh-Ubaid) [inline integration_tests/string_26.f90:7]: Should we use indentation here? @certik what do you suggest?

The fortran backend generated code uses indentation. If we use indentation in the above test source, then we can easily compare this test source code with the fortran backend generated code using some diff comparison tool like https://www.diffchecker.com/. 

I think diff comparison tools like https://www.diffchecker.com/ (or others) get confused when one of the code is indented and the other is without indentation.

Comment (certik) [reply integration_tests/string_26.f90:7]: I would change our backend to not do indentation here. It's a matter of preference in the end, so we can make it configurable like we did for the AST->Fortran backend. I think it's better to keep the Fortran indentation to the left, to save horizontal space, which is scarce.

**Review by certik — APPROVED** [review]

**Comment by certik** [comment]
Conflicts have to be resolved.

**Comment by khushi-411** [comment]
Done!

File: integration_tests/string_26.f90 [hunk 3]
```diff
@@ -0,0 +1,8 @@
+program string26
```
Comment (Shaikh-Ubaid) [inline integration_tests/string_26.f90:1]:
```suggestion
program string_26
```

Comment (khushi-411) [reply integration_tests/string_26.f90]: updated, thanks for your helpful comments, @Shaikh-Ubaid!

**Comment by khushi-411** [comment]
Thanks, @certik!

**Pull request merged**